<!DOCTYPE html>
<head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <title>Pairwise Sequentially Markovian Coalescent Model</title>
  <meta name="author" content="Bruno Vieira">

  <link rel="stylesheet" href="vendors/revealjs/css/reveal.css">
  <link rel="stylesheet" href="vendors/revealjs/css/print/reveal-like-pdf.css">
  <link rel="stylesheet" href="vendors/Font-Awesome/css/font-awesome.min.css">
  <link rel="stylesheet" href="theme/style.css">
</head>

<body>

  <div class="reveal">
    <div class="slides">
      
<section data-state=front>
<h2 id="pairwise-sequentially-markovian-coalescent-model">Pairwise Sequentially Markovian Coalescent Model</h2>
<p><a href="//bmpvieira.com/labjolly2014a" target="_blank">bmpvieira.com/labjolly2014a</a></p>
<p>Bruno Vieira | <i class="fa fa-twitter"></i> <a href="//twitter.com/bmpvieira" target="_blank">@bmpvieira</a></p>
<p><img src="img/Queen_Mary,_University_of_London_logo.svg" alt="QMUL"></p>
</section>

<section data-state=title>
<h2 id="disclaimer">Disclaimer</h2>
<p><img class="fragment" style="width: 60%;" alt="afra-stack" src="img/afra_stack.png" /></p>
</section>

<section data-state=q>
<h2 id="what-is-the-psmc-">What is the PSMC?</h2>
</section>

<section>
<h3 id="psmc-result">PSMC result</h3>
<p><img class="stretch" style="max-width: 100%;" src="img/psmc.png" alt="psmc" /></p>
<p><a href="//dx.doi.org/10.1038/nature10231" target="_blank">Li, 2011</a></p>
</section>

<section data-state=q>
<h2 id="what-is-it-used-for-">What is it used for?</h2>
</section>

<section>
<h3 id="effect-of-climate-changes">Effect of climate changes</h3>
<p><img src="img/climate-changes.jpg" /></p>
<div style="display: block;">
<!-- <img style="max-width: 20%; float: left; padding-right: 2%;" src="img/polar-bear.jpg" alt="http://en.wikipedia.org/wiki/File:Polar_Bear_-_Alaska.jpg" /> -->

<p> <span style="font-size:.5em;"><a href="//dx.doi.org/10.1073/pnas.1210506109" target="_blank">[Miller, 2012]</a></span></small></p>
</div>


</section>

<section>
<h3 id="discover-unexpected-population-bottlenecks">Discover unexpected population bottlenecks</h3>
<p><img style="max-width: 75%;" src="img/dog-psmc.png" /></p>
<p><small><a href="//dx.doi.org/10.1371/journal.pgen.1004016" target="_blank">[Freedman, 2014]</a></span></small></p>
</section>

<section>
<h3 id="detect-the-time-of-divergence-between-populations">Detect the time of divergence between populations</h3>
<p><img class="stretch" style="max-width: 100%;" src="img/psmc.png" alt="psmc" /></p>
<p><small><a href="//dx.doi.org/10.1038/nature10231" target="_blank">[Li, 2011]</a></span></small></p>
</section>

<section data-state=q>
<h2 id="what-i-ve-been-doing-">What I&#39;ve been doing?</h2>
</section>

<section>
<h3 id="reproducing-published-results-to-master-psmc">Reproducing published results to master PSMC</h3>
<p><img src="img/human.jpg" alt="https://upload.wikimedia.org/wikipedia/commons/1/11/Uomo_Vitruviano.jpg" style="max-width: 25%" /><img src="img/wolf.jpg" alt="https://upload.wikimedia.org/wikipedia/commons/f/ff/Wolf%2C_voor_de_natuur%2C_Saxifraga_-_Jan_Nijendijk.5097.jpg" style="max-width: 39%" /></p>
<ul>
<li><a href="//dx.doi.org/10.1038/nature10231" target="_blank">Li, 2011</a></li>
<li><a href="//dx.doi.org/10.1371/journal.pgen.1004016" target="_blank">Freedman, 2014</a></li>
</ul>
</section>

<section data-state=q>
<h2 id="what-i-intend-to-do-">What I intend to do?</h2>
</section>

<section>
<h3 id="use-psmc-to-answer-some-evolutionary-questions">Use PSMC to answer some evolutionary questions</h3>
</section>

<section data-state=q>
<h2 id="is-the-effective-population-size-in-solitary-insects-social-">Is the effective population size in solitary insects &gt; social?</h2>
</section>

<section>
<h3 id="experimental-design">Experimental design</h3>
<p>Run PSMC across a wide range of social insects and their solitary relatives</p>
<p><img src="img/WeaverAntDefense.JPG" alt="From: http://upload.wikimedia.org/wikipedia/commons/8/85/WeaverAntDefense.JPG" style="max-width: 25%" />
<img src="img/Honeybee_landing_on_milkthistle02.jpg" alt="From: http://upload.wikimedia.org/wikipedia/commons/e/e0/Honeybee_landing_on_milkthistle02.jpg" style="max-width: 23%" /></p>
<p><img src="img/Coptotermes_formosanus_shiraki_USGov_k8204-7.jpg" alt="From: http://upload.wikimedia.org/wikipedia/commons/d/d3/Coptotermes_formosanus_shiraki_USGov_k8204-7.jpg" style="max-width: 16%" />
<img src="img/Bush_Cockroach.jpg" alt="From: http://upload.wikimedia.org/wikipedia/commons/c/ca/Bush_Cockroach.jpg" style="max-width: 32%" /></p>
</section>

<section data-state=q>
<h2 id="how-does-it-work-theory-">How does it work (theory)?</h2>
</section>

<section>
<h3 id="psmc">PSMC</h3>
<p><img class="stretch" alt="http://www.troll.me/images/futurama-fry/what-sorcery-is-this.jpg" src="img/what-sorcery-is-this.jpg" /></p>
<p>Short answer: <span class="fragment">I don&#39;t know!</span></p>
<p>Long answer: <span class="fragment">I can try to explain...</span></p>
</section>

<section>
<h3 id="history">History</h3>
<h4 id="-a-href-http-dx-doi-org-10-1016-0040-5809-75-90020-9-watterson-1975-a-"><a href="http://dx.doi.org/10.1016/0040-5809(75)90020-9">Watterson, 1975</a></h4>
<blockquote>
<p>Generating the genealogical relationship between k sampled sequences from a population with constant size N and no recombination</p>
</blockquote>
<p><small><a href="http://dx.doi.org/10.1006/tpbi.1998.1403">[Wiuf and Hein, 1999]</a></small></p>
</section>

<section>
<h3 id="coalescent-theory">Coalescent theory</h3>
<p><a href='http://dx.doi.org/10.1016/0304-4149(82)90011-4'>Kingman, 1982</a></p>
<p><img style="width: 70%;" alt="coalescent theory" src="img/coalescence.png"></p>
<p><small>
<a href='http://dx.doi.org/10.1186/1471-2105-15-3'>[Yang, 2014]</a>
</small></p>
</section>

<section>
<h3 id="coalescent-theory-with-recombination">Coalescent theory with recombination</h3>
<p><span class="fragment">
<a href="http://dx.doi.org/10.1016/0040-5809(83)90013-8">Hudson, 1983</a>
</span></p>
<div class="fragment">
<p>
<a href="http://dx.doi.org/10.1089/cmb.1996.3.479">Griffiths and Marjoram, 1997</a>
</p>
<p>Ancestral Recombination Graph (ARG)</p>

<img style="width: 25%;" alt="arg" src="img/arg.png">

</div>

</section>

<section>
<p><img style="width: 55%;" alt="arg" src="img/arg.png"></p>
</section>

<section>
<h3 id="spatial-approach">Spatial approach</h3>
<p><a href="http://dx.doi.org/10.1006/tpbi.1998.1403">Wiuf and Hein, 1999</a></p>
<p><small>Along the chromosome instead of back in time</small></p>
<p><img style="width: 80%;" alt="spatial" src="img/spatial.png"></p>
<p><small>
<a href="http://dx.doi.org/10.1186/1471-2156-7-16">Marjoram, 2006</a>
</small></p>
</section>

<section>
<h3 id="recombination-events-that-occur-on-the-arg">Recombination events that occur on the ARG</h3>
<div style="font-size: 0.8em;">
<p>1. Recombination in ancestral material;</p>
<p>2. Recombination in non-ancestral material that has ancestral material to both sides;</p>
<p>3. Recombination in non-ancestral material that has ancestral material only to the left;</p>
<p>4. Recombination in non-ancestral material that has ancestral material only to the right;</p>
<p>5. Recombination in an individual that carries no ancestral material.</p>

</div>

<p><small>
<a href="http://dx.doi.org/10.1186/1471-2156-7-16">Marjoram, 2006</a>
</small></p>
</section>

<section>
<h3 id="recombination-events-that-occur-on-the-arg">Recombination events that occur on the ARG</h3>
<p>Spatial approach by <a href="http://dx.doi.org/10.1006/tpbi.1998.1403">Wiuf and Hein, 1999</a></p>
<div style="font-size: 0.8em;">
<p>1. Recombination in ancestral material;</p>
<p>2. Recombination in non-ancestral material that has ancestral material to both sides;</p>
<p>3. Recombination in non-ancestral material that has ancestral material only to the left;</p>
<p><span style="text-decoration: line-through;">4. Recombination in non-ancestral material that has ancestral material only to the right;</span></p>
<p><span style="text-decoration: line-through;">5. Recombination in an individual that carries no ancestral material.</span></p>

</div>

</section>

<section>
<h3 id="sequential-markov-coalescent">Sequential Markov Coalescent</h3>
<p>SMC <a href="http://dx.doi.org/10.1098%2Frstb.2005.1673">[McVean &amp; Cardin, 2005]</a></p>
<p>SMC&#39; <a href="http://dx.doi.org/10.1186/1471-2156-7-16">[Marjoram &amp; Wall], 2006</a></p>
<p><img style="width: 45%;" src="img/smc.png" /></p>
<p><small>
<a href='http://dx.doi.org/10.1016/0304-4149(82)90011-4'>A) SMC; B) SMC&#39;</a> <a href='http://dx.doi.org/10.1186/1471-2105-15-3'>[Yang, 2014]</a>
</small></p>
</section>

<section>
<h3 id="markov-chain">Markov chain</h3>
<p><img alt="http://upload.wikimedia.org/wikipedia/commons/2/2b/Markovkate_01.svg" style="width: 45%;" src="img/Markov.svg" /></p>
</section>

<section>
<h3 id="pairwise-sequential-markov-coalescent">Pairwise Sequential Markov Coalescent</h3>
<p><a href="//dx.doi.org/10.1038/nature10231" target="_blank">Li, 2011</a></p>
<p><img style="width: 65%;" src="img/psmc-model.png" /></p>
</section>

<section>
<h3 id="demographic-inference-using-composite-approximate-likelihood-dical-">Demographic inference using composite approximate likelihood (diCal)</h3>
<p><a href="http://dx.doi.org/10.1534/genetics.112.149096">Sheehan, 2013</a></p>
<blockquote>
<p><span style="font-size:.7em; line-height:0;">&quot;New coalescent-based method that can efficiently infer population size changes from multiple genomes, providing access to a new store of information about the recent past.&quot;</span></p>
</blockquote>
</section>

<section>
<p><img style="width: 60%;" src="img/psmc-phylo.png" /></p>
<p><small>
<a href="http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf">http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf</a>
</small></p>
</section>

<section>
<p><img style="width: 70%;" src="img/psmc-phylo-demo.png" /></p>
<p><small>
<a href="http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf">http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf</a>
</small></p>
</section>

<section>
<p><img style="width: 80%;" src="img/psmc-phylo-specie.png" /></p>
<p><small>
<a href="http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf">http://simons.berkeley.edu/sites/default/files/docs/1213/slidesmailund.pdf</a>
</small></p>
</section>

<section data-state=q>
<h2 id="how-does-it-work-practice-">How does it work (practice)?</h2>
</section>

<section>
<h3 id="install">Install</h3>
<p><span class="fragment">$ git clone <a href="https://github.com/lh3/psmc.git">https://github.com/lh3/psmc.git</a></span></p>
<p><span class="fragment">$ cd psmc</span></p>
<p><span class="fragment">$ make</span></p>
<p><span class="fragment">$ cd utils</span></p>
<p><span class="fragment">$ make</span></p>
</section>

<section>
<h3 id="run">Run</h3>
<div style="font-size:.8em; text-align: left;">

<span class="fragment">$ utils/fq2psmcfa diploid.fq.gz &gt; diploid.psmcfa</span>
<br>
<span class="fragment">$ psmc -o diploid.psmc diploid.psmcfa</span>
<!-- <span class="fragment">$ psmc -N25 -t15 -r5 -p "4+25*2+4+6" \ <br> -o diploid.psmc diploid.psmcfa</span>
 --><br>
<span class="fragment">$ utils/psmc_plot.pl diploid diploid.psmc</span>

</div>

<p><br></p>
<!-- <span class="fragment">-N maximum number of iterations [30]</span>
<br>
<span class="fragment">-t FLOAT maximum 2N0 coalescent time [15]</span>
<br>
<span class="fragment">-r initial theta/rho ratio [4]</span>
<br>
<span class="fragment">-p STR pattern of parameters [4+5*3+4]</span> -->

</section>

<section>
<h3 id="parameters">Parameters</h3>
<pre><code class="lang-bash">Usage:   psmc [options] input.txt

Options: -p STR      pattern of parameters [4+5*3+4]
         -t FLOAT    maximum 2N0 coalescent time [15]
         -N INT      maximum number of iterations [30]
         -r FLOAT    initial theta/rho ratio [4]
         -c FILE     CpG counts generated by cntcpg [null]
         -o FILE     output file [stdout]
         -i FILE     input parameter file [null]
         -T FLOAT    initial divergence time; -1 to disable [-1]
         -b          bootstrap (input be preprocessed with split_psmcfa)
         -S          simulate sequence
         -d          perform decoding
         -D          print full posterior probabilities
</code></pre>
</section>

<section>
<h3 id="parameters">Parameters</h3>
<pre><code class="lang-bash">Usage:   psmc_plot.pl [options] &lt;out.prefix&gt; &lt;in.psmc&gt;

Options: -u FLOAT   absolute mutation rate per nucleotide [2.5e-08]
         -s INT     skip used in data preparation [100]
         -X FLOAT   maximum generations, 0 for auto [0]
         -x FLOAT   minimum generations, 0 for auto [10000]
         -Y FLOAT   maximum popsize, 0 for auto [0]
         -m INT     minimum number of iteration [5]
         -n INT     take n-th iteration (suppress GOF) [20]
         -M titles  multiline mode [null]
         -f STR     font for title, labels and tics [Helvetica,16]
         -g INT     number of years per generation [25]
         -w INT     line width [4]
         -P STR     position of the keys [right top]
         -T STR     figure title [null]
         -N FLOAT   false negative rate [0]
         -S         no scaling
         -L         show the last bin
         -p         convert to PDF (with epstopdf)
         -R         do not remove temporary files
         -G         plot grid
</code></pre>
</section>

<section>
<h3 id="thank-you-">Thank you!</h3>
<p><img style="max-width: 15%;" src="img/bmpvieira.png" alt"Bruno Vieira"/>
Bruno Vieira | <i class="fa fa-twitter"></i> <a href="//twitter.com/bmpvieira" target="_blank">@bmpvieira</a></p>
<p><img style="max-width: 15%;" src="img/yannick.jpg" alt"Yannick Wurm" />
Yannick Wurm | <i class="fa fa-twitter"></i> <a href="//twitter.com/yannick__" target="_blank">@yannick__</a></p>
<p><a href="//bmpvieira.com/labjolly14a" target="_blank">bmpvieira.com/labjolly14a</a></p>
<p><small>
Â© 2014 <a href="//bmpvieira.com" target="_blanl">Bruno Vieira</a> <a href="//creativecommons.org/licenses/by/4.0/deed.en_US" target="_blank">CC-BY 4.0</a>
</small></p>
</section>

<section>

</section>

    </div>
  </div>

<script src="vendors/jquery-1.11.0.min.js"></script>

<!-- reveal -->
<script src="vendors/revealjs/lib/js/head.min.js"></script>
<script src="vendors/revealjs/js/reveal.js"></script>
<script>
  Reveal.initialize({

    // Display controls in the bottom right corner
    controls: false,

    // Display a presentation progress bar
    progress: true,

    // Display a subtle timer bar (time is in minutes)
    timeRemaining: 10,

    // Push each slide change to the browser history
    history: true,

    // Enable keyboard shortcuts for navigation
    keyboard: true,

    // Enable the slide overview mode
    overview: true,

    // Vertical centering of slides
    center: false,

    // Loop the presentation
    loop: false,

    // Change the presentation direction to be RTL
    rtl: false,

    // Number of milliseconds between automatically proceeding to the
    // next slide, disabled when set to 0, this value can be overwritten
    // by using a data-autoslide attribute on your slides
    autoSlide: 0,

    // Enable slide navigation via mouse wheel
    mouseWheel: false,

    // Apply a 3D roll to links on hover
    rollingLinks: false,

    // Transition style
    transition: 'linear', // default/cube/page/concave/zoom/linear/fade/none

    dependencies: [
        // Cross-browser shim that fully implements classList - https://github.com/eligrey/classList.js/
        { src: 'vendors/revealjs/lib/js/classList.js', condition: function() { return !document.body.classList; } },

        // Interreveal/pret Markdown in <section> elements
    { src: 'vendors/revealjs/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
    { src: 'vendors/revealjs/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); }, callback: function() { baconDemo() } },

        // Syntareveal/x highlight for <code> elements
          { src: 'vendors/revealjs/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },

          // Zoom reveal/in and out with Alt+click
          { src: 'vendors/revealjs/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },

          // Speakreveal/er notes
          { src: 'vendors/revealjs/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },

          // Remotreveal/e control your reveal.js presentation using a touch device
          // { src: 'reveal/plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; } }
          ]

          });
    </script>
</body>

